awx_collection/plugins/modules/subscriptions.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+# (c) 2019, John Westcott IV <john.westcott.iv@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1', 'status': ['preview'], 'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: get_subscriptions
+author: "John Westcott IV (@john-westcott-iv)"
+short_description: Get subscription list
+description:
+    - Get subscriptions available to  Automation Platform Controller. See
+      U(https://www.ansible.com/tower) for an overview.
+options:
+    username:
+      description:
+        - Red Hat or Red Hat Satellite username to get available subscriptions.
+        - The credentials you use will be stored for future use in retrieving renewal or expanded subscriptions
+      required: True
+      type: str
+    password:
+      description:
+        - Red Hat or Red Hat Satellite password to get available subscriptions.
+        - The credentials you use will be stored for future use in retrieving renewal or expanded subscriptions
+      required: True
+      type: str
+extends_documentation_fragment: awx.awx.auth
+'''
+
+RETURN = '''
+subscriptions:
+    description: dictionary containing information about the subscriptions
+    returned: If login succeeded
+    type: dict
+'''
+
+EXAMPLES = '''
+- name: Get subscriptions
+  get_subscriptions:
+    username: "my_username"
+    password: "My Password"
+'''
+
+from ..module_utils.controller_api import ControllerAPIModule
+
+
+def main():
+
+    module = ControllerAPIModule(
+        argument_spec=dict(
+            username=dict(type='str', required=True),
+            password=dict(type='str', no_log=True, required=True),
+        ),
+    )
+
+    json_output = {'changed': False}
+
+    # Check if Tower is already licensed
+    post_data = {
+        'subscriptions_password': module.params.get('password'),
+        'subscriptions_username': module.params.get('username'),
+    }
+    json_output['subscriptions'] = module.post_endpoint('config/subscriptions', data=post_data)['json']
```

Comment:
We should probably filter here, this should work
```
    subscription_data = module.post_endpoint('config/subscriptions', data=post_data)['json']

    subscription_aap_list = []
    for subscription in subscription_data:
        if 'Red Hat Ansible Automation Platform' in subscription['subscription_name']:
            subscription_aap_list.append(subscription)
    json_output['subscriptions'] = subscription_aap_list
    module.exit_json(**json_output)
```

Reply:
I added a way to fully filter this but didn't default anything. LMK what you think about that. Also added an example of filtering.